### PR TITLE
bpo-40147: Move the check for duplicate keywords to the compiler

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue40147

### CODE REVIEW

**Comment by pablogsal** [comment]
As a bonus now we print the actual keyword that is repeated.

Before:
```
❯ python lel.py                       
  File "lel.py", line 1
    f(1, x=2, *(3, 4), x=5)
                       ^
SyntaxError: keyword argument repeated
```

After:
```
❯ ./python lel.py  
  File "/home/pablogsal/github/python/master/lel.py", line 1
    f(1, x=2, *(3, 4), x=5)
                       ^
SyntaxError: keyword argument repeated: x
```

**Comment by pablogsal** [comment]
CC: @lysnikolaou

File: Python/compile.c [hunk 1]
```diff
@@ -4049,6 +4049,34 @@ maybe_optimize_method_call(struct compiler *c, expr_ty e)
...
+    int nkeywords = asdl_seq_LEN(keywords);
+    for (int i = 0; i < nkeywords; i++) {
+        keyword_ty key = ((keyword_ty)asdl_seq_GET(keywords, i));
+        if (key->arg == NULL) {
+            continue;
+        }
+        for (int j = 0; j < nkeywords; j++) {
+            if (i == j) {
+                continue;
+            }
```
Comment (lysnikolaou) [inline Python/compile.c]: Couldn't this just be `for (int j = i+1; j < keywords; j++)`? This way we could also omit the `if (i ==j )` check.

Comment (pablogsal) [reply Python/compile.c]: Done!

**Review by gvanrossum — APPROVED** [review]
Nice! That was simpler than I expected.

**Pull request merged**